src/api/api.js

```diff
@@ -9,6 +9,30 @@ app.use(cors());
 const port = process.env.API_PORT || 3000;
 const DATA_DIR = process.cwd() + '/data';
 
+
+
+// getprecomputeddata?platform=uniswapv2&span=1
+app.get('/api/getprecomputeddata', async (req, res, next) => {
+    try {
+        console.log('received getprecomputeddata request', req);
+        const platform = req.query.platform;
+        const span = Number(req.query.span);
+
+        if(!span) {
+            res.status(400).json({error: 'span required'});
+            next();
+        }
+        const fileName = `concat-${span}d.json`;
+        const returnObject = JSON.parse(fs.readFileSync(`${DATA_DIR}/precomputed/${platform}/${fileName}`));
```

Comment:
need to add check/error with fs.existsSync() and return http 404 if the file does not exists

Reply:
also please use path.join(DATA_DIR, 'precomputed', platform, filename) to make sure the API works fines if hosted on windows (it will not but...)